src/workerd/api/gpu/gpu-device.c++

```diff
@@ -287,8 +289,59 @@ GPUDevice::createComputePipeline(GPUComputePipelineDescriptor descriptor) {
     }
   }
 
+  return kj::mv(desc);
+}
+
+jsg::Ref<GPUComputePipeline>
+GPUDevice::createComputePipeline(GPUComputePipelineDescriptor descriptor) {
+  wgpu::ComputePipelineDescriptor desc =
+      parseComputePipelineDescriptor(descriptor);
   auto pipeline = device_.CreateComputePipeline(&desc);
   return jsg::alloc<GPUComputePipeline>(kj::mv(pipeline));
 }
 
+jsg::Promise<jsg::Ref<GPUComputePipeline>>
+GPUDevice::createComputePipelineAsync(GPUComputePipelineDescriptor descriptor) {
+  wgpu::ComputePipelineDescriptor desc =
+      parseComputePipelineDescriptor(descriptor);
+
+  struct Context {
+    kj::Own<kj::PromiseFulfiller<jsg::Ref<GPUComputePipeline>>> fulfiller;
+    AsyncTask task;
+  };
+  auto paf = kj::newPromiseAndFulfiller<jsg::Ref<GPUComputePipeline>>();
+  // This context object will old information for the callback, including the
+  // fullfiller to signal the caller with the result, and an async task that
+  // will ensure the device's Tick() function is called periodically. It will be
+  // deallocated at the end of the callback function.
+  auto ctx = new Context{kj::mv(paf.fulfiller), AsyncTask(async_)};
+
+  device_.CreateComputePipelineAsync(
```

Comment:
Looking at the API model for this, I think I'd prefer that `CreateComputePipelineAsync(...)` would return a `kj::Promise<T>` that is fulfilled by the passed in lambda returning a `T`. That would simplify things a bit by not requiring the `kj::newPromiseAndFulfiller` to be created and passed in via the `ctx`.

Reply:
We can address this later. I'll work up an example of what I mean in a later PR.